eotsmanager/cmd/eotsd/daemon/keys.go

```diff
@@ -134,22 +141,22 @@ func saveKeyNameMapping(cmd *cobra.Command, keyName string) (*types.BIP340PubKey
 	}
 
 	if len(rpcListener) > 0 {
-		client, err := eotsclient.NewEOTSManagerGRpcClient(rpcListener, "")
+		eotsClient, err := eotsclient.NewEOTSManagerGRpcClient(rpcListener, "")
 		if err != nil {
 			return nil, err
 		}
 
-		kr, err := eotsmanager.InitKeyring(clientCtx.HomeDir, clientCtx.Keyring.Backend(), strings.NewReader(""))
-		if err != nil {
-			return nil, fmt.Errorf("failed to init keyring: %w", err)
-		}
-
-		eotsPk, err := eotsmanager.LoadBIP340PubKeyFromKeyName(kr, keyName)
+		eotsPk, err := eotsmanager.LoadBIP340PubKeyFromKeyName(clientCtx.Keyring, keyName)
 		if err != nil {
 			return nil, fmt.Errorf("failed to get public key for key %s: %w", keyName, err)
 		}
 
-		if err := client.SaveEOTSKeyName(eotsPk.MustToBTCPK(), keyName); err != nil {
+		if err := eotsClient.SaveEOTSKeyName(eotsPk.MustToBTCPK(), keyName); err != nil {
+			// ignore the err, keyring will handle it
+			if errors.Is(err, store.ErrDuplicateEOTSKeyName) {
+				return eotsPk, nil
+			}
```

Comment:
We probably need to check if the key is the same as the key with the same key name. If not, we probably should reject